### PR TITLE
Backtrace dumping support for Windows

### PR DESCRIPTION
**Pull request by zach2good**
Spent a while trying to remember how to be a Windows developer to give us backtrace logging for Win platforms. Eventually tripped over the exact snippet we needed, and put in additional logging support :D

**DUMPS TO LOGS**

image

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by zach2good** [comment]
Tested with:
```cpp
    throw;
    std::vector<int> v; v[55] = 10;
    std::vector<int> w; auto die = w.at(50);
    *((unsigned int*)0) = 0xDEAD;
```

Properly logs to console and file 👍

**Review by claywar — APPROVED** [review]

**Comment by zach2good** [comment]
As I've started testing this morning, I've started encountering crashes in Lua at startup that weren't there last night... 

Hold off on merge until I've had time to investigate

File: src/common/kernel.cpp [hunk 1]
```diff
@@ -143,6 +148,208 @@ static void dump_backtrace()
...
+    // https://love2d.org/forums/viewtopic.php?t=84336
+    switch (pExceptionInfo->ExceptionRecord->ExceptionCode)
+    {
+        // exceptions NOT documented by Microsoft
+        case 0xE24C4A02 : // Magic exception number from LuaJIT: Ignore!
+            return EXCEPTION_CONTINUE_SEARCH;
+
+        default:
+            break;
+    }
```
Comment (zach2good) [inline src/common/kernel.cpp:252]: Found the issue, and it's pretty interesting!

LuaJIT throws exceptions and swallows them as part of its regular operation. So if we hook onto them... yeah.

This block tells the handler to skip over any exceptions from LuaJIT. (We rely on exceptions and all bad things in Lua being swallowed anyway, so nothing is being lost here)

This also explains what all those random silent exceptions at startup and runtime are!

**Comment by zach2good** [comment]
I'm happy with this for now, I'll look into that other library (and the Linux side) in another PR

File: src/common/kernel.cpp (hunk 1 above)
Comment (claywar) [reply src/common/kernel.cpp:252]: Found another one that was encountered when saving a lua script with some bad copy-pasta, `0xE24C4A03`.  Adding it to the switch resolved.

**Pull request merged**